docs/reference/commandline/build.md

```diff
@@ -251,3 +252,61 @@ flag](/reference/run/#specifying-custom-cgroups).
 Using the `--ulimit` option with `docker build` will cause each build step's 
 container to be started using those [`--ulimit`
 flag values](/reference/run/#setting-ulimits-in-a-container).
+
+Sometimes building an image will require access to certain environment variables
```

Comment:
@mapuri This section goes into too much detail about the `ARG` and `ENV` interactions --- it really belongs in the Dockerfile reference. I'll be updating my comments shortly.  Also, need to mention the `inspect` command for finding arguments.

---> https://gist.github.com/moxiegirl/176c608aee68da99887c

When `docker build` is run with the `--cgroup-parent` option the containers
used in the build will be run with the [corresponding `docker run`
flag](/reference/run/#specifying-custom-cgroups). 

Using the `--ulimit` option with `docker build` will cause each build step's 
container to be started using those [`--ulimit`
flag values](/reference/run/#setting-ulimits-in-a-container).

Currently, you can use `ENV` instructions in a Dockerfile to define variable
values. These values persist in the final, built image. However, often
persistence is not what you want. Users want to specify variables differently depending on which host they use an image on. 

A good example is `http-proxy` or source versions for pulling intermediate
files. The `ARG` instruction lets Dockefile developers define values that users
can set at build-time using the  `---build-arg` flag:

    $ docker build --build-arg HTTP_PROXY=http://10.20.30.2:1234 .

This flag allows you to pass the build-time variables that are
accessed like regular environment variables in the `RUN` primitive of the
Dockerfile. Also, these values don't persist in the intermediate or final images like `ENV` values do.

To find out the build arguments available in a image, use the [`docker inspect`](/reference/inspect) command. For detailed information on using `ARG` and `ENV` instructions, see the [Dockerfile reference}(/reference/builder).